src/tree/localProject/ConfigGroupTreeItem.ts

```diff
@@ -3,37 +3,101 @@
-import { LocalProjectTreeItem } from "./LocalProjectTreeItem";
+import { EnvironmentTreeItem } from "../EnvironmentTreeItem";
+import { GitHubConfigTreeItem } from "./ConfigTreeItem";
 
-export class ConfigGroupTreeItem extends AzExtParentTreeItem {
-    public static contextValue: string = 'azureStaticConfigGroup';
-    public contextValue: string = ConfigGroupTreeItem.contextValue;
-    public readonly label: string = localize('config', 'Configuration');
-    public parent: LocalProjectTreeItem;
+export type BuildConfig = 'app_location' | 'api_location' | 'app_artifact_location' | 'output_location';
 
-    public constructor(parent: LocalProjectTreeItem) {
+export class GitHubConfigGroupTreeItem extends AzExtParentTreeItem {
+    public static contextValue: string = 'azureStaticGitHubConfigGroup';
+    public contextValue: string = GitHubConfigGroupTreeItem.contextValue;
+    public readonly label: string = localize('gitHubConfig', 'GitHub Configuration');
+    public parent: EnvironmentTreeItem;
+    public yamlFilePath: string;
+    public buildConfigs: Map<BuildConfig, string>;
+
+    public constructor(parent: EnvironmentTreeItem, yamlFilePath: string, buildConfigs: Map<BuildConfig, string>) {
         super(parent);
+        this.yamlFilePath = yamlFilePath;
+        this.buildConfigs = buildConfigs;
+        this.id = `${parent.id}-${this.yamlFilePath}`;
+    }
+
+    public static async createGitHubConfigGroupTreeItems(parent: EnvironmentTreeItem): Promise<GitHubConfigGroupTreeItem[]> {
+        if (parent.localProjectPath && parent.inWorkspace) {
+            const treeItems: GitHubConfigGroupTreeItem[] = [];
+            const workflowsDir: string = join(parent.localProjectPath, '.github/workflows');
+            const yamlFiles: string[] = await readdir(workflowsDir);
+
+            for (const yamlFile of yamlFiles) {
+                if (yamlFile.endsWith('.yml')) {
+                    const yamlFilePath: string = join(workflowsDir, yamlFile);
+                    const contents: string = (await readFile(yamlFilePath)).toString();
+
+                    if (/Azure\/static-web-apps-deploy/.test(contents)) {
+                        /* eslint-disable @typescript-eslint/no-unsafe-call, @typescript-eslint/no-explicit-any, @typescript-eslint/no-unsafe-assignment, @typescript-eslint/no-unsafe-member-access, no-prototype-builtins */
+                        const parsedYaml: any = await parse(contents);
+                        const buildConfigs: Map<BuildConfig, string> = new Map();
+
+                        for (const job of <any[]>Object.values(parsedYaml.jobs)) {
+                            for (const step of <any[]>Object.values(job['steps'])) {
+                                if (step.hasOwnProperty('id') && step['id'] === 'builddeploy') {
+                                    if (!step.hasOwnProperty('with') || !step['with'].hasOwnProperty('api_location') || !step['with'].hasOwnProperty('app_location')) {
```

Comment:
And then if it's verified, we can start extracting the values.

Also, could you make this a function that accepts the buildConfig parameter and returns the value of the configuration?  There's a decent amount of repeat code here.